frontend/src/components/forms/EditLog.tsx

```diff
@@ -202,6 +209,7 @@ const EditLog = ({
 
   const initializeValues = () => {
     // set state variables
+    getBuildingsOptions();
```

Comment:
so instead of calling this here, you should call this in `LogRecordsTable.tsx`. the issue is that the state doesn't update in time, so when the building ID from the log record is being matched with these options, they don't exist yet (when you close and re-open the options have loaded ands appear)

there's a function there called `getLogEntryOptions` that you can pop this in and then pass it as a prop to this component which should resolve things